### PR TITLE
Support newer Phonopy yaml versions

### LINKED ISSUE #276: Born charge / dielectric data keywords missed from some phonopy.yaml
I have a phonopy.yaml file with the section

```
nac:
  born_effective_charge:
  - # 1 ...
  ...
  dielectric_constant:
  - [  ...
  unit_conversion_factor: 14.4000
```

but Euphonic looks for the keys 'born', 'dielectric', 'nac_factor' and misses them.

Presumably the Phonopy output format has changed at some point; we should support these (better and more verbose!) keys as well.

### PR DESCRIPTION
**Pull request by ajjackson**
Fixes #276

Since version 2.18 Phonopy has moved some keywords into an `nac` block.

For now we retain compatibility with older versions.

A unit test is included, with data from a recent version of Phonopy (2.19.1).

I noticed this parser uses a lot of exception handling; explicit checks for key existence seem safer if that is what we are trying to determine. For now I just refactored the function being worked on; perhaps other parts can wait until we have the walrus, which is nice for this sort of thing. ( #278 )

### CODE REVIEW

File: euphonic/readers/phonopy.py [hunk 1]
```diff
@@ -543,28 +543,38 @@ def _extract_summary(filename: str, fc_extract: bool = False
...
+            summary_dict['born'] = summary_object.get('born_effective_charge')
+            summary_dict['dielectric'] = summary_object.get(
+                'dielectric_constant')
+
+        # Cast arrays and drop keys if data not found
+        for key in ('nac_factor', 'born', 'dielectric'):
+            if summary_dict[key] is None:
+                del summary_dict[key]
+            elif key in ('born', 'dielectric'):
+                summary_dict[key] = np.array(summary_dict[key])
```
Comment (oerc0122) [inline euphonic/readers/phonopy.py:577]: For the sake off saving a loop through 2 elements in a non-performance critical application, I find a second loop more explicit.
```suggestion

        for key in ('born', 'dielectric'):
            summary_dict[key] = np.array(summary_dict[key])
```

**Review by oerc0122 — APPROVED** [review]
Assuming the test data are OK.

File: euphonic/readers/phonopy.py (hunk 1 above)
Comment (ajjackson) [reply euphonic/readers/phonopy.py:577]: This will create array(None) if the keys were not present in the file.

Comment (ajjackson) [reply euphonic/readers/phonopy.py:577]: Ah wait, actually we already deleted it so we will get a KeyError?

Comment (oerc0122) [reply euphonic/readers/phonopy.py:577]: Adding
```
if key in summary_dict:
```
?
It was a suggestion as I find it describes what's going on better, definitely not required.

Comment (ajjackson) [reply euphonic/readers/phonopy.py:577]: Like this?

```python
        # Drop NAC keys if no data was found
        for key in ('nac_factor', 'born', 'dielectric'):
            if summary_dict[key] is None:
                del summary_dict[key]

        for key in ('born', 'dielectric'):
            if key in summary_dict:
                summary_dict[key] = np.array(summary_dict[key])
```

**Comment by ajjackson** [comment]
I'm quite puzzled by this test failure because
- it does superficially appear to be relevant as it relates to finding Born charge data
- but I can reproduce it on `master`
- but CI passed on `master` a few weeks ago

**Comment by ajjackson** [comment]
I will fire off CI tests on `master` and see if they still pass... I have some conda environments where this test passes and some where it fails, and the one where it passes is too complicated to bisect :sweat_smile:

**Pull request merged**